packages/core/src/dynamoDb/commands/searchDynamoDbTables.ts

```diff
@@ -0,0 +1,109 @@
+ */
+
+import * as vscode from 'vscode'
+import * as nls from 'vscode-nls'
+import fs from '../../shared/fs/fs'
+import DynamoDB from 'aws-sdk/clients/dynamodb'
+import { Wizard } from '../../shared/wizards/wizard'
+import { telemetry } from '../../shared/telemetry/telemetry'
+import { DataQuickPickItem } from '../../shared/ui/pickerPrompter'
+import { DynamoDbClient } from '../../shared/clients/dynamoDbClient'
+import { toArrayAsync } from '../../shared/utilities/collectionUtils'
+import { CancellationError } from '../../shared/utilities/timeoutUtils'
+import { RegionSubmenu, RegionSubmenuResponse } from '../../shared/ui/common/regionSubmenu'
+
+const localize = nls.loadMessageBundle()
+
+/**
+ * "Search DynamoDb Tables Command"
+ *
+ */
+export async function searchDynamoDbTables(source: string, dbData?: { regionName: string }): Promise<void> {
+    await telemetry.dynamodb_openTable.run(async span => {
+        const wizard = new SearchDynamoDbTablesWizard(dbData)
+        span.record({ dynamoDbResourceType: 'table', source: source })
+        const response = await wizard.run()
+        if (!response) {
+            throw new CancellationError('user')
+        }
+        const tableOutput = await getItemsFromTable(response.submenuResponse.region, response.submenuResponse.data)
+
+        const output = `${JSON.stringify(tableOutput.Items, undefined, 4)}\n`
+        const filePath = process.env['HOME'] + '/Documents/dynamoDbItems.json'
+        await fs.writeFile(filePath, output)
+        const uri = vscode.Uri.parse(filePath)
+        await prepareDocument(uri)
+    })
+}
+
+export async function prepareDocument(uri: vscode.Uri) {
+    try {
+        const doc = await vscode.workspace.openTextDocument(uri)
+        await vscode.window.showTextDocument(doc, { preview: false })
+        await vscode.languages.setTextDocumentLanguage(doc, 'json')
+    } catch (err) {
+        if (CancellationError.isUserCancelled(err)) {
+            throw err
+        }
+    }
+}
+
+export interface SearchDynamoDbTablesWizardResponse {
+    submenuResponse: RegionSubmenuResponse<string>
+    filterPattern: string
+}
+
+export function createRegionSubmenu() {
+    return new RegionSubmenu(
+        getTablesFromRegion,
+        { title: localize('AWS.dynamoDb.searchTables.TableTitle', 'Select a table') },
+        { title: localize('AWS.dynamoDb.searchTables.regionPromptTitle', 'Select Region for DynamoDb') },
+        'DynamoDb Tables'
+    )
+}
+
+async function getTablesFromRegion(regionCode: string): Promise<DataQuickPickItem<string>[]> {
+    const client = new DynamoDbClient(regionCode)
+    const dynamoDbTables = await dynamoDbTablesToArray(client.getTables())
+    const options = dynamoDbTables.map<DataQuickPickItem<string>>(tableName => ({
+        label: tableName,
+        data: tableName,
+    }))
+    return options
+}
+
+async function dynamoDbTablesToArray(dynamoDbTables: AsyncIterableIterator<string>): Promise<string[]> {
```

Comment:
It looks like this is very similiar to `logGroupsToArray` is there any way we can generalize the two? and maybe put it in collection utils or something like that

Reply:
Only three lines of code are common. I believe it might be over-engineering.